components-api/src/main/java/pl/touk/nussknacker/engine/api/editor/Editor.java

```diff
@@ -0,0 +1,36 @@
+package pl.touk.nussknacker.engine.api.editor;
+
+import java.lang.annotation.ElementType;
+import java.lang.annotation.Repeatable;
+import java.lang.annotation.Retention;
+import java.lang.annotation.RetentionPolicy;
+import java.lang.annotation.Target;
+import java.time.temporal.ChronoUnit;
+
+@Target(ElementType.PARAMETER)
+@Retention(RetentionPolicy.RUNTIME)
+@Repeatable(Editors.class)
+public @interface Editor {
+
+    EditorType type();
+
+    /**
+     * This field should be used only with {@link EditorType#FIXED_VALUES_EDITOR}
+     */
+    LabeledExpression[] possibleValues() default {};
+
+    /**
+     * This field should be used only with {@link EditorType#DURATION_EDITOR} or {@link EditorType#PERIOD_EDITOR}
+     */
+    ChronoUnit[] timeRangeComponents() default {};
+
+    /**
+     * This field should be used only with {@link EditorType#DICT_EDITOR}
+     */
+    String dictId() default "";
+
+    /**
+     * Whether the editor will be displayed at first on the UI.
+     */
+    boolean isMainEditor() default false;
```

Comment:
main is very generic name that can be interpreted in multiple ways. By looking to comment, the only thing that you want say by this field is that editor should be first on the list of visible editors. If so, let's rename this field.
Why orders of editors is not enough? It is not staithforward that if you declared some editor as first, it will be first, if you declared as second, it will be second etc?

Reply:
Done - now order matters